### PR TITLE
drivers: Clear the registered waiters for polling when closing filep

### PR DESCRIPTION
**Pull request by jeongarmy**
Check if this file is registered in a list of waiters for polling.
It it is, the used slot should be cleared.
Otherwise, a invalid pollfd remains in a list and this slot is not available forever.

### CODE REVIEW

File: os/drivers/gpio/gpio.c [hunk 1]
```diff
@@ -583,6 +586,19 @@ static int gpio_close(FAR struct file *filep)
 		return ret;
 	}
 
+#ifndef CONFIG_DISABLE_POLL
+	/* Check if this file is registered in a list of waiters for polling.
+	 * It it is, the used slot should be cleared.
```
Comment (jeongchanKim) [inline os/drivers/gpio/gpio.c:591]: Please modify `It` to `If`.

File: os/drivers/gpio/gpio.c [hunk 2]
```diff
@@ -559,6 +559,9 @@ static int gpio_close(FAR struct file *filep)
 	irqstate_t flags;
 	bool closing;
 	int ret;
+#ifndef CONFIG_DISABLE_POLL
+	int i;
```
Comment (jeongchanKim) [inline os/drivers/gpio/gpio.c:563]: You'd better rename this for clarification.

File: os/drivers/gpio/gpio.c [hunk 3]
```diff
@@ -583,6 +586,19 @@ static int gpio_close(FAR struct file *filep)
 		return ret;
 	}
 
+#ifndef CONFIG_DISABLE_POLL
+	/* Check if this file is registered in a list of waiters for polling.
+	 * It it is, the used slot should be cleared.
+	 * Otherwise, a invalid pollfd remains in a list and this slot is not available forever.
```
Comment (jeongchanKim) [inline os/drivers/gpio/gpio.c:592]: a invalid -> an invalid

File: os/drivers/lwnl/lwnl.c [hunk 4]
```diff
@@ -144,20 +144,39 @@ static int lwnl_open(struct file *filep)
...
-
 static int lwnl_close(struct file *filep)
 {
 	LWNL_ENTER;
 	struct inode *inode = filep->f_inode;
 	struct lwnl_upperhalf_s *upper = inode->i_private;
 	int ret = OK;
+#ifndef CONFIG_DISABLE_POLL
+	FAR struct lwnl_open_s *opriv = &upper->ln_open;
+	int i;
```
Comment (jeongchanKim) [inline os/drivers/lwnl/lwnl.c:155]: Same as above.

File: os/drivers/lwnl/lwnl.c [hunk 5]
```diff
@@ -144,20 +144,39 @@ static int lwnl_open(struct file *filep)
...
 
-
 static int lwnl_close(struct file *filep)
 {
 	LWNL_ENTER;
 	struct inode *inode = filep->f_inode;
 	struct lwnl_upperhalf_s *upper = inode->i_private;
 	int ret = OK;
+#ifndef CONFIG_DISABLE_POLL
+	FAR struct lwnl_open_s *opriv = &upper->ln_open;
```
Comment (Taejun-Kwon) [inline os/drivers/lwnl/lwnl.c:154]: it's not private data of upper, renaming should be done.

Comment (jeongarmy) [reply os/drivers/lwnl/lwnl.c:154]: I don't understand your comment. Could you explain more? It is used internally.

Comment (Taejun-Kwon) [reply os/drivers/lwnl/lwnl.c:154]: It's not major comment, but priv being used for private context data, usually.
But ln_open is defined explicitly, it's not good naming.
For example, priv is dev_a's priv pointer in below..
```
 struct dev_a {
  int type;
  int value;
  void *priv;
}

struct lowhalf_a {
   int size;
   int length;
}

struct dev_a *g_dev;

struct a initalize_driverA(lowhalf_a *dev) {
  g_dev->priv = dev;
  register_driver(path, &dev_aops, 0666, g_dev);
  
}

// dev_aops 
static int dev_a_open(FAR struct file *filep)
{
	FAR struct inode *inode = filep->f_inode;
	FAR struct dev_a *dev = inode->i_private;
	FAR struct lowhalf_a *priv = dev->priv;
	...
}
``` 
```

Comment (Taejun-Kwon) [reply os/drivers/lwnl/lwnl.c:154]: To sum up, priv is being used for iNode's private data(upper) or private context data(lower) of private data(upper)

File: os/drivers/gpio/gpio.c [hunk 6]
```diff
@@ -583,6 +586,19 @@ static int gpio_close(FAR struct file *filep)
 		return ret;
 	}
 
+#ifndef CONFIG_DISABLE_POLL
+	/* Check if this file is registered in a list of waiters for polling.
```
Comment (deepaksrma) [inline os/drivers/gpio/gpio.c:590]: please give the space below and up from the comments.
it enhance the readability.

File: os/drivers/lwnl/lwnl.c (hunk 5 above)
Comment (jeongarmy) [reply os/drivers/lwnl/lwnl.c:154]: Thank you for explanation. I think it is just a data of upper struct, not private context data(lower) as you said. I will rename it.

**Review by jeongchanKim — APPROVED** [review]
LGTM

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**